docker/package/tezos_voting_wizard.py

```diff
@@ -0,0 +1,559 @@
+                config_contents, 'NODE_RPC_ENDPOINT="(.*)"', "http://localhost:8732"
+            )
+            self.config["baker_alias"] = search_baking_service_config(
+                config_contents, 'BAKER_ADDRESS_ALIAS="(.*)"', "baker"
+            )
+
+    def get_tezos_client_options(self):
+        return (
+            "--base-dir "
+            + self.config["client_data_dir"]
+            + " --endpoint "
+            + self.config["node_rpc_addr"]
+        )
+
+    def fill_voting_period_info(self):
+        info = get_proc_output(
+            f"sudo -u tezos {suppress_warning_text} tezos-client "
+            f"{self.config['tezos_client_options']} show voting period"
+        ).stdout
+
+        self.config["amendment_phase"] = (
+            re.search(b'Current period: "(\w+)"', info).group(1).decode("utf-8")
+        )
+        self.config["proposal_hashes"] = [
+            phash.decode() for phash in re.findall(protocol_hash_regex, info)
+        ]
+
+    def process_proposal_period(self):
+        self.query_step(get_proposal_period_hash(self.config["proposal_hashes"]))
+
+        hash_to_submit = self.config["chosen_hash"]
+        if hash_to_submit == "Specify new proposal hash":
+            self.query_step(new_proposal_query)
+            hash_to_submit = self.config["new_proposal_hash"]
+
+        result = get_proc_output(
+            f"sudo -u tezos {suppress_warning_text} tezos-client {self.config['tezos_client_options']} "
+            f"submit proposals for {self.config['baker_alias']} {hash_to_submit}"
+        )
+
+        if result.returncode == 0:
+            if yes_or_no("Submission successful! Choose again? <Y/n> ", "yes"):
+                self.process_proposal_period()
+                return
+        else:
+            print()
+
+            if re.search(b"[Ii]nvalid proposal", result.stderr) is not None:
+                print(color("The submitted proposal hash is invalid.", color_red))
+                print("Check your custom submitted proposal hash and try again.")
+                self.process_proposal_period()
+                return
+            elif re.search(b"Unauthorized proposal", result.stderr) is not None:
+                print(
+                    color(
+                        "Cannot submit because of an unauthorized proposal.", color_red
+                    )
+                )
+                print("This means you are not present in the voting listings.")
+            elif re.search(b"Too many proposals", result.stderr) is not None:
+                print(
+                    color(
+                        "Cannot submit because of too many proposals submitted.",
+                        color_red,
+                    )
+                )
+                print("This means you have already submitted more than 20 proposals.")
+            # No other "legitimate" proposal error ('empty_proposal', 'unexpected_proposal')
```

Comment:
Do I understand correctly that 'unexpected_proposal' happens when the proposal hash is malformed and this is caught by the regex check in our code?

Reply:
No, that should be the `Invalid proposal` one, and our regex can't catch all invalid hashes, it mostly just checks the length.
I'm not sure what the `unexpected_proposal` is for 😕 The docs mention it's when the voting is not in the 'proposal' period, but in that case, `tezos-client` reports "Not in a proposal period".